File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/guild/tax/GuildFightLeaveRequestMessage.scala





















// Generated on 05/03/2013 10:34:51
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GuildFightLeaveRequestMessage extends MessageDeserializer {
  val messageId = 5715

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GuildFightLeaveRequestMessage.messageId

    val taxCollectorId: Int
    val characterId: Int

    def serialize(buf: Buffer) {
      buf.writeInt(taxCollectorId)
      buf.writeInt(characterId)
    }

  }

  protected def create(taxCollectorId: Int, characterId: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val taxCollectorId = buf.readInt()
    val characterId = buf.readInt()

    create(taxCollectorId, characterId)
  }

}

object GuildFightLeaveRequestMessage extends GuildFightLeaveRequestMessage {
  case class MessageImpl(taxCollectorId: Int, characterId: Int) extends Message
  type Target = MessageImpl

  override def create(taxCollectorId: Int, characterId: Int) = MessageImpl(taxCollectorId, characterId)

  def apply(taxCollectorId: Int, characterId: Int) = create(taxCollectorId, characterId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GuildFightLeaveRequestMessage.java


// Generated on 05/08/2013 19:37:52
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GuildFightLeaveRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5715;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int taxCollectorId;
    public int characterId;
    
    public GuildFightLeaveRequestMessage() { }
    
    public GuildFightLeaveRequestMessage(int taxCollectorId, int characterId) {
        this.taxCollectorId = taxCollectorId;
        this.characterId = characterId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(taxCollectorId);
        buf.writeInt(characterId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        taxCollectorId = buf.readInt();
        characterId = buf.readInt();
        if (characterId < 0)
            throw new RuntimeException("Forbidden value on characterId = " + characterId + ", it doesn't respect the following condition : characterId < 0");
    }
    
}
